File: Source/ARTFallbackHosts.m
#import "ARTFallbackHosts.h"

#import "ARTDefault+Private.h"
#import "ARTClientOptions+Private.h"

@implementation ARTFallbackHosts

+ (nullable NSArray<NSString *> *)hostsFromOptions:(ARTClientOptions *)options {
    if (options.fallbackHosts) {
        return options.fallbackHosts;
    }

    if (options.hasEnvironmentDifferentThanProduction) {
        return [ARTDefault fallbackHostsWithEnvironment:options.environment];
    }
    if (options.hasCustomRestHost || options.hasCustomRealtimeHost || options.hasCustomPort || options.hasCustomTlsPort) {
        return nil;
    }
    return [ARTDefault fallbackHosts];
}

@end

File: Sources/AblySwift/ARTFallbackHosts.swift
import Foundation

// swift-migration: original location ARTFallbackHosts.h, line 7 and ARTFallbackHosts.m, line 6
internal class ARTFallbackHosts: NSObject {
    
    // swift-migration: original location ARTFallbackHosts.h, line 9 and ARTFallbackHosts.m, line 8
    internal class func hosts(from options: ARTClientOptions) -> [String]? {
        if let fallbackHosts = options.fallbackHosts {
            return fallbackHosts
        }
        
        // swift-migration: original location ARTFallbackHosts.m, line 15
        if options.fallbackHostsUseDefault {
            return ARTDefault.fallbackHosts()
        }
        
        if options.hasEnvironmentDifferentThanProduction {
            return ARTDefault.fallbackHosts(withEnvironment: options.environment!)
        }
        
        if options.hasCustomRestHost || options.hasCustomRealtimeHost || options.hasCustomPort || options.hasCustomTlsPort {
            return nil
        }
        
        return ARTDefault.fallbackHosts()
    }
}
